discovery_engine_core/providers/src/newscatcher.rs

```diff
@@ -12,8 +12,200 @@
+        }
+    }
+
+    pub fn from_endpoint(endpoint: RestEndpoint) -> Arc<dyn NewsProvider> {
+        Arc::new(Self { endpoint })
+    }
+}
+
+fn max_age_to_date_string(max_age_days: usize) -> String {
+    // (lj): This _could_ overflow if we specified trillions of days, but I don't
+    // think that's worth guarding against.
+    let days = max_age_days as i64;
+
+    let from = Utc::today() - chrono::Duration::days(days);
+    from.format("%Y/%m/%d").to_string()
+}
+
+#[allow(clippy::unnecessary_wraps)]
+fn to_generic_articles(articles: Vec<Article>) -> Result<Vec<GenericArticle>, Error> {
+    let articles = articles
+        .into_iter()
+        .flat_map(GenericArticle::try_from)
+        .collect();
+    Ok(articles)
+}
+
+#[async_trait]
+impl NewsProvider for NewscatcherNewsProvider {
+    async fn query_news(&self, request: &NewsQuery<'_>) -> Result<Vec<GenericArticle>, Error> {
+        let response = self
+            .endpoint
+            .get_request::<Response, _>(|mut query| {
+                query
+                    .append_pair("page_size", &request.page_size.to_string())
+                    .append_pair("page", &request.page.to_string());
+
+                if !request.excluded_sources.is_empty() {
+                    query.append_pair("not_sources", &request.excluded_sources.join(","));
+                }
+
+                query.append_pair("sort_by", "relevancy");
+                append_market(&mut query, request.market);
+                query.append_pair("q", &request.filter.build());
+
+                if let Some(days) = &request.max_age_days {
+                    query.append_pair("from", &max_age_to_date_string(*days));
+                }
+            })
+            .await?;
+
+        to_generic_articles(response.articles)
+    }
+}
+
+pub struct NewscatcherHeadlinesProvider {
+    endpoint: RestEndpoint,
+}
+
+impl NewscatcherHeadlinesProvider {
+    /// Create a new provider.
+    pub fn new(endpoint_url: Url, auth_token: String) -> Self {
+        Self {
+            endpoint: RestEndpoint::new(endpoint_url, auth_token),
+        }
+    }
+
+    pub fn from_endpoint(endpoint: RestEndpoint) -> Arc<dyn HeadlinesProvider> {
+        Arc::new(Self { endpoint })
+    }
+}
+
+#[async_trait]
+impl HeadlinesProvider for NewscatcherHeadlinesProvider {
+    async fn query_headlines(
+        &self,
+        request: &HeadlinesQuery<'_>,
+    ) -> Result<Vec<GenericArticle>, Error> {
+        let response = self
+            .endpoint
+            .get_request::<Response, _>(|mut query| {
+                query
+                    .append_pair("page_size", &request.page_size.to_string())
+                    .append_pair("page", &request.page.to_string());
+
+                if !request.excluded_sources.is_empty() {
+                    query.append_pair("not_sources", &request.excluded_sources.join(","));
+                }
+
+                query.append_pair("sort_by", "relevancy");
```

Comment:
:wrench: 
```suggestion

```

Reply:
Mmh, is this really obsolete? It appears that, in the current `main` branch, we still use this parameter as well. Can you elaborate?